### PR TITLE
Qemu external snap test

### PR DESCRIPTION
**Pull request by mihalicyn**
test for https://github.com/canonical/lxd/pull/13742

### CODE REVIEW

File: tests/qemu-external-vm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+fi
+
+# Install LXD
+install_lxd
+
+# Configure LXD
+lxd init --auto
+
+# Get the source
+git clone "https://github.com/canonical/lxd-pkg-snap.git" -b "latest-edge"
```
Comment (simondeziel) [inline tests/qemu-external-vm:26]:
```suggestion
git clone --depth=1 "https://github.com/canonical/lxd-pkg-snap.git" -b "latest-edge"
```
Will be a tad faster ;)

File: tests/qemu-external-vm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# check that qemu-external connect hook works properly
+journalctl --quiet --no-hostname --no-pager --boot=0 --unit=snap.lxd.daemon.service | grep "Setting up external QEMU snap integration"
+
+# check that qemu instance driver is activated and is in "external" mode
+lxc query /1.0 | jq '.environment.driver' | grep "qemu"
+lxc query /1.0 | jq '.environment.driver_version' | grep "(external)"
+
+# check that connection with mesa-2404 snap works
+serverPID="$(lxc query /1.0 | jq .environment.server_pid)"
+cat /proc/${serverPID}/environ | grep LIBGL_DRIVERS
```
Comment (simondeziel) [inline tests/qemu-external-vm:53]:
```suggestion
grep -awF LIBGL_DRIVERS /proc/${serverPID}/environ
```

**Comment by tomponline** [comment]
Looks like some shellcheck issues.

File: tests/qemu-external-vm (hunk 1 above)
Comment (mihalicyn) [reply tests/qemu-external-vm]: Fixed! Thanks!

File: tests/qemu-external-vm (hunk 2 above)
Comment (mihalicyn) [reply tests/qemu-external-vm]: Fixed! Thanks!

**Comment by mihalicyn** [comment]
Run gh extension install actions/gh-actions-cache
  gh extension install actions/gh-actions-cache
  gh actions-cache delete "cache-20240716" --confirm
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    PURGE_LXD: 1
    GH_TOKEN: ***
Error: Resource not accessible by integration

**Comment by mihalicyn** [comment]
It works in the fork repo. So, no problems with cache anymore, hopefully.

**Review by tomponline — CHANGES_REQUESTED** [review]
Please can you address or silence the shellcheck issues.

**Comment by mihalicyn** [comment]
> Please can you address or silence the shellcheck issues.

all of them are obsolete

File: tests/qemu-external-vm [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  mv "${LXD_QEMU_SNAP_FILE_NAME}" "${CACHE_PATH}/"
+  echo "${LXD_QEMU_SNAP_HASH_SUM}" > "${LXD_QEMU_SNAP_HASHSUM_FILE_PATH}"
+fi
+
+SNAP_TO_INSTALL_PATH="${CACHE_PATH}/${LXD_QEMU_SNAP_FILE_NAME}"
+
+# debug
+ls -la "${CACHE_PATH}/"
+
+snap install "${SNAP_TO_INSTALL_PATH}" --devmode # TODO: change to --dangerous
```
Comment (tomponline) [inline tests/qemu-external-vm:61]: Please address this TODO :)

Comment (tomponline) [reply tests/qemu-external-vm:61]: ah ok thanks!

Comment (mihalicyn) [reply tests/qemu-external-vm]: Have fixed! Thanks!

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
Thanks!

**Pull request merged**